File: src/main/java/com/example/hexagonal/domain/Product.java
package com.example.hexagonal.domain;

import jakarta.persistence.*;
import java.io.Serializable;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "products")
@Getter
@Setter
public class Product implements Serializable {

  private static final long serialVersionUID = 1L;

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(name = "pname")
  private String name;

  private Integer stock;
}

File: kotlin/src/main/kotlin/com/example/hexagonal/domain/Product.kt
package com.example.hexagonal.domain

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.io.Serializable

@Entity
@Table(name = "products")
class Product(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long,

    @Column(name = "pname")
    var name: String,

    var stock: Int? = 0,
) : Serializable {
    companion object {
        const val DEFAULT_ID = 1L
        const val DEFAULT_NAME = "Foo"
        const val DEFAULT_STOCK = 1

        fun aProduct() = Product(id = DEFAULT_ID, name = DEFAULT_NAME, stock = DEFAULT_STOCK)
    }
}
